### PR TITLE
OBAListView iOS 14 list

### LINKED ISSUE #377: Header label being smushed when image is used
Header taking up too much vertical space. In trip view controller, this causes the list's cells to be completely covered and unreadable. As far as I can tell, this only occurs when the text size is set to an accessibility value.

image

### PR DESCRIPTION
**Pull request by ualch9**
## Changelog
- Replaces custom recreation of UITableView cells and supplementary views in favor of UIListContentConfiguration -- #377
- Removes SwipeCellKit in favor of UIListContentConfiguration.
- Uses `insetGrouped` visual style for most lists (not bookmarks, bookmark cells look too space-constrained in the new style).
- OBAListViewCollapsibleSectionsDelegate is broken now -- #392

## Shiny new visual look
| ooooo | ahhhhh |
| -- | -- |
| IMG_3448 |  IMG_3447 |

## References
Fixes #377

Opened:
#392
#393
#398 
#399

### CODE REVIEW

**Comment by ualch9** [comment]
just noticed memory leaks. needs investigation but i think it's caused by listview caching data snapshots.

image

**Comment by ualch9** [comment]
memory leak not related to this pr, opened #394

File: OBAKit/Bookmarks/TripBookmarkCell.swift [hunk 1]
```diff
@@ -130,8 +130,6 @@ final class TripBookmarkTableCell: OBAListViewCell {
     override init(frame: CGRect) {
         super.init(frame: frame)
 
-        fixiOS13AutoLayoutBug()
```
Comment (aaronbrethorst) [inline OBAKit/Bookmarks/TripBookmarkCell.swift:133]: 🎉

File: OBAKit/Controls/ListView/OBAListView.swift [hunk 2]
```diff
@@ -212,17 +164,74 @@ public class OBAListView: UICollectionView, UICollectionViewDelegate, SwipeColle
...
     fileprivate func createLayout() -> UICollectionViewLayout {
-        return UICollectionViewCompositionalLayout { section, _ -> NSCollectionLayoutSection? in
-            return self.diffableDataSource.snapshot().sectionIdentifiers[section].sectionLayout
+        return UICollectionViewCompositionalLayout { [unowned self] section, environment -> NSCollectionLayoutSection? in
+            let sectionModel = self.lastDataSourceSnapshot[section]
+
+            var configuration = sectionModel.configuration.listConfiguration()
+            configuration.headerMode = sectionModel.hasHeader ? .firstItemInSection : .none
+
+            // #398 - list view separators always appears on iOS 14.4
```
Comment (aaronbrethorst) [inline OBAKit/Controls/ListView/OBAListView.swift:186]: I like that you opened a task for this!

File: OBAKit/Controls/ListView/OBAListView.swift [hunk 3]
```diff
@@ -237,31 +246,47 @@ public class OBAListView: UICollectionView, UICollectionViewDelegate, SwipeColle
...
-            self.diffableDataSource.apply(snapshot, animatingDifferences: animated) { [weak self] in
-                guard let self = self else { return }
-                DispatchQueue.main.async {
-                    self.obaDelegate?.didApplyData(self)
-                }
-            }
+            diffableDataSource.apply(sectionSnapshot, to: section, animatingDifferences: false)
         }
+
+//        DispatchQueue.main.async {
```
Comment (aaronbrethorst) [inline OBAKit/Controls/ListView/OBAListView.swift:282]: delete me!

File: OBAKit/Controls/ListView/Rows/OBAListRowConfiguration.swift [hunk 4]
```diff
@@ -32,6 +32,15 @@ public struct OBAListRowConfiguration: OBAContentConfiguration, Hashable, Equata
         case detailButton
         case disclosureIndicator
         case none
+
+        var cellAccessory: UICellAccessory? {
+            switch self {
+            case .checkmark: return .checkmark(displayed: .always, options: .init(tintColor: ThemeColors.shared.brand))
+            case .detailButton: return .label(text: "i") // TODO: this
```
Comment (aaronbrethorst) [inline OBAKit/Controls/ListView/Rows/OBAListRowConfiguration.swift:39]: ?

File: OBAKit/Stops/Sections/StopArrival/StopArrivalCell.swift [hunk 5]
```diff
@@ -66,6 +65,7 @@ final class StopArrivalCell: OBAListViewCell {
     }
 
     func showNudge() {
-        showSwipe(orientation: .right)
+        // TODO: me!
```
Comment (aaronbrethorst) [inline OBAKit/Stops/Sections/StopArrival/StopArrivalCell.swift:68]: todo!

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
This looks like a great improvement! I noticed a couple of outstanding TODOs that you'd added that look like they still need to be taken care of. If you want to add issues for those and have me merge this, np. Otherwise let me know when they're wrapped up and I'll get this merged.

File: OBAKit/Stops/Sections/StopArrival/StopArrivalCell.swift (hunk 5 above)
Comment (ualch9) [reply OBAKit/Stops/Sections/StopArrival/StopArrivalCell.swift:68]: opened #401

**Review by aaronbrethorst — APPROVED** [review]
:shipit: nice work

**Pull request merged**